server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -46,6 +54,29 @@
  * @opensearch.internal
  */
 public final class RemoteStoreRefreshListener implements ReferenceManager.RefreshListener {
```

Comment:
can we add UTs for this change and cover the four cases of `isRetry` and `shouldRetry` ?